src/project.c

```diff
@@ -772,6 +772,53 @@ static gboolean update_config(const PropertyDialogElements *e, gboolean new_proj
 		}
 		g_free(locale_path);
 	}
+
+	/* create filename path if it doesn't exist */
+	if ((err_code = utils_is_file_writable(locale_filename)) != 0) {
+
+		/* only create path if filename ends with '.geany' */
+		char *dot = strrchr(locale_filename, '.');
+		if (dot != NULL && strcmp(dot, ".geany") == 0) {
+
+			char path_delimiter = '/';
+
+#ifdef G_OS_WIN32
+            path_delimiter = '\\';
+#endif
+
+			/* get last path_delimiter so we know the path */
+			int i;
+			int last_path_delimiter;
+			for (i = 0; locale_filename[i] != '\0'; ++i) {
+				if (locale_filename[i] == path_delimiter) {
+					last_path_delimiter = i;
+				}
+			}
+
+			/* get substring containing only the filename path */
+			gchar *filename_path = g_malloc((last_path_delimiter + 2) * sizeof(char));
+			strncpy(filename_path, locale_filename, last_path_delimiter + 1);
+			filename_path[last_path_delimiter + 1] = '\0';
```

Comment:
If I understand correctly the intention here, I believe these lines could be reduced to:

```c
gchar *filename_path = g_path_get_dirname(locale_filename);
```

GLib makes C quite a bit less painful :wink:

Reply:
+1 Just FTR https://developer.gnome.org/glib/stable/index.html

Reply:
Thank you very much @codebrainz !

The code now is much simpler! I tested on Ubuntu and it worked.

And probably `g_path_get_dirname` already takes care of operating system path delimiter.